### PR TITLE
add a README on go mod/go sum differences

### PR DESCRIPTION
**Pull request by tuminoid**
Add a README on go.mod and go.sum difference, and how the files should or should not be used for vulnerability scanning, or licensing.

### CODE REVIEW

**Review by lentzi90 — APPROVED** [review]
Excellent! Thanks for this!
/approve

File: security/go-mod-and-go-sum.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+- "Each Go module is defined by a go.mod file that describes the module’s
+   properties, including its dependencies on other modules and on versions of
+   Go." - [go.dev](https://go.dev/doc/modules/gomod-ref),
+   [go.dev](https://go.dev/ref/mod)
+
+- "The go.mod file lists the specific versions of the dependencies that your
+   project uses." -
+   [FreeCodeCamp](https://www.freecodecamp.org/news/golang-environment-gopath-vs-go-mod/#:~:text=sum%20file.-,The%20go.,all%20projects%20on%20your%20system.)
+
+- "No, go.sum is not a lock file. The go.mod files in a build provide
```
Comment (kashifest) [inline security/go-mod-and-go-sum.md:21]: The same line is added in go.sum section as well, intentional ?

**Review by kashifest — APPROVED** [review]
Thanks for the excellent writeup, one comment in line. lgtm

File: security/go-mod-and-go-sum.md (hunk 1 above)
Comment (tuminoid) [reply security/go-mod-and-go-sum.md:21]: Yes, it talks about both in a meaningful way. For go.mod it talks about reproducability and for go.sum it talks about not being a lock file, and is then followed up by a longer snippet.

**Pull request merged**